src/main/java/net/mcreator/ui/blockly/BlocklyJavascriptBridge.java

```diff
@@ -177,8 +177,15 @@ yield openDataListEntrySelector(w -> ElementUtil.loadDataListAndElements(w, type
 	 */
 	private String[] openDataListEntrySelector(Function<Workspace, List<DataListEntry>> entryProvider, String type) {
 		String[] retval = new String[] {"", L10N.t("blockly.extension.data_list_selector.no_entry")};
-		String title = L10N.t("dialog.selector." + type + ".title");
-		String message = L10N.t("dialog.selector." + type + ".message");
+		String title, message;
+		if (DataListLoader.getCache().containsKey(type)) {
+			String dataList = L10N.t("dialog.generator_selector.coverage." + type);
+			title = L10N.t("dialog.selector.title", dataList);

```

Comment:
I would prefer if you keep these separate translation keys as it is right now. The plural form may not play well with some languages, as well as uppercase text that may not work well if some languages can use "{0} entry selection" word order.

You can keep the "Please select an entry from the list:" translation "optimization", though

Reply:
So just that for the message and <kbd>Data list entry selection</kbd> for the title?

Reply:
Or the other way around, whatever you think fits better as I don't have an opinion about that.

Your suggestion is fine

> So just that for the message and Data list entry selection for the title?

Reply:
This is what I did initially, no?

Reply:
So am I supposed to rewrite all these strings for the selector dialog?

Reply:
Or just keep those strings as they were. I can try to reword it:

The way it was before this PR was perfectly fine (better than the current state in this PR). If you goal was to "optimize" amount of translation keys (generally not good idea as languages may need different way of saying things that are not compatible with such optimizations), at best you can eg. make frame title common to all of dialogs, but still keep message as it is in label dialog.

Reply:
>Or the other way around, whatever you think fits better as I don't have an opinion about that.

So selector still takes coverage-related strings for the title. If that's not fine, I have prepared a local commit that adds all removed titles back.

Reply:
> So selector still takes coverage-related strings for the title.

No. 

> not good idea as languages may need different way of saying things that are not compatible with such optimizations

> If that's not fine, I have prepared a local commit that adds all removed titles back.

Yes, or:

> make frame title common to all of dialogs, but still keep message as it is (as it is in master, not in this PR) in label dialog.

Reply:
> So selector still takes coverage-related strings for the title.

No. 

> not good idea as languages may need different way of saying things that are not compatible with such optimizations

`{0} data list entry selections` may not be acceptable word order for some other language

> If that's not fine, I have prepared a local commit that adds all removed titles back.

Yes, or:

> make frame title common to all of dialogs, but still keep message as it is (as it is in master, not in this PR) in label dialog.

Reply:
>`{0} data list entry selections` may not be acceptable word order for some other language

Eh, I supposed such languages could e.g. put the list name in quotes (`Selection of entry from data list "{0}"`).

Reply:
> generally not good idea as languages may need different way of saying things that are not compatible with such optimizations

We should allow languages to use things freely. I have told the two options that are acceptable already

Reply:
✔️ Fixed